carousel/core/calculations.py

```diff
@@ -171,6 +180,15 @@ def __init__(self):
         else:
             #: parameter file
             self.param_file = None
+        # XXX: Hack to get PR#68 done
+        if not self.parameters:
```

Comment:
After reading the PR 10 times and Carousel code (+ some blogs on metaclasses and __new__()) I'm starting to get a small understanding of things.

I thought that ``mcs.set_param_file_or_parameters()`` and the few lines above would guarantee that self.parameters would never be None? Or is it in case the user does not specify any parameter?
Does it also check for missing fields? For instance if the user forget to specify 'frequency'.